src/dataflow/components/dataflow-program.tsx

```diff
@@ -614,6 +628,24 @@ export class DataflowProgram extends BaseComponent<IProps, IState> {
     });
   }
 
+  private getDatasetName() {
+    let datasetName: string | undefined;
+    this.programEditor.nodes.forEach((n: Node) => {
+      if (n.name === "Data Storage") {
+        datasetName = n.data.datasetName as string | undefined;
+      }
+    });
+    return datasetName;
+  }
+
+  private hasDataStorage() {
+    return this.getNodeCount("Data Storage") > 0;
+  }
+
+  private hasRelay() {
+    return this.getNodeCount("Relay") > 0;
+  }
```

Comment:
Thanks for consolidating the checks for a relay or data storage block.  Much cleaner.